mono/mini/interp/transform.c

```diff
@@ -4539,8 +4539,12 @@ generate_code (TransformData *td, MonoMethod *method, MonoMethodHeader *header,
 					} else {
 						if (mint_type (m_class_get_byval_arg (klass)) == MINT_TYPE_VT)
 							interp_add_ins (td, MINT_NEWOBJ_VTST_FAST);
-						else
+						else {
+							// Increase maximum stack and then restore.
+							move_stack (td, (td->sp - td->stack) - csignature->param_count, 1);
+							move_stack (td, (td->sp - td->stack) - csignature->param_count, -1);
```

Comment:
This is very ugly. Just add a new macro. Something like : ENSURE_STACK_SPACE(sp, size) which updates max_stack_height if necessary

Reply:
Ugly: somewhat agreed.
My defense: Using mechanisms I've succeeded with before.
Function ok, I hope.
Macros are for syntactic forms, like `if_error_goto` and `if_error_return`, or "generics" (`max`).
Rarely needed is "force inclining", trust the compiler.
Granted, this might juts be `td->max_something += 1`, once I look closer (again, reusing mechanism..allowed a certain ignorance/laziness).

Reply:
I tried. We'll what CI thinks..